spec/controllers/news_controller_spec.rb

```diff
@@ -0,0 +1,101 @@
+require 'rails_helper'
+
+RSpec.describe NewsController, type: :controller do
+  let(:news) { create(:valid_news) }
+
+  describe 'GET #index' do
+    it 'render index template' do
+      get :index
+      is_expected.to render_template(:index)
+    end
+  end
+
+  describe 'GET #new' do
+    it 'render new template' do
+      get :new
+      is_expected.to render_template(:new)
+    end
+  end
+
+  describe 'GET #show' do
+    it 'render show template' do
+      get :show, params: { id: news.id }
+      is_expected.to render_template(:show)
+    end
+  end
+
+  describe 'POST #create' do
+    let(:user) { create(:user) }
+    before(:each) do
+      allow(controller).to receive(:current_user).and_return(user)
+    end
+
+    it 'increases number of news' do
+      expect do
+        post :create, params: { news: attributes_for(:valid_news) }
+      end.to change(News, :count).by(1)
+    end
+
+    it 'redirects to index action' do
+      post :create, params: { news: attributes_for(:valid_news) }
+      is_expected.to redirect_to(action: :index)
+    end
+
+    it 'valid news and sets flash after it' do
+      post :create, params: { news: attributes_for(:valid_news) }
+      is_expected.to set_flash
+    end
+
+    it 'render new template if news is invalid' do
+      post :create, params: { news: attributes_for(:invalid_news) }
+      is_expected.to render_template(:new)
+    end
+
+    it 'sets flash if news is invalid' do
+      post :create, params: { news: attributes_for(:invalid_news) }
+      is_expected.to set_flash
+    end
+  end
+
+  describe 'PUT #update' do
+    let(:char2) { Faker::Lorem.characters(2) }
+    let(:char3) { Faker::Lorem.characters(3) }
+    it 'news with valid data' do
+      put :update, params: { id: news.id, news: attributes_for(:valid_news, title: :char3) }
+      is_expected.to redirect_to(action: :index)
+    end
+
+    it 'news with valid data and sets flash' do
+      put :update, params: { id: news.id, news: attributes_for(:valid_news, title: :char3) }
+      is_expected.to set_flash
+    end
+
+    it 'can not change news with invalid data' do
+      put :update, params: { id: news.id, news: attributes_for(:valid_news, title: :char2) }
+      is_expected.to render_template(nil)
+    end
+
+    it 'can not change news with invalid data and sets flash' do
+      put :update, params: { id: news.id, news: attributes_for(:valid_news, title: :char2) }
+      is_expected.to set_flash
+    end
+  end
+
+  describe 'DELETE #destroy' do
+    it 'reduces number of news' do
```

Comment:
something is wrong with this test.

you say, 'reduces number of news' but the check you do is `to_not change(News, :count)` BUT it should change. it should reduce a number of news

Reply:
In the 4 line i create news
In the 87 line i delete this news
Thats why the number of news doesnt change

Reply:
It should change! Otherwise your controller does not work

Reply:
done